### PR TITLE
I added .has functions, and added .is and .has functions for ```#```

### PR DESCRIPTION
**Pull request by NHQ**
There was a bunch of functions of isTypeId

I needed _has_TypeID, to scan whole messages for link types.

I then added a set of functions to test links for the "tag" sigil: `#`

### CODE REVIEW

**Comment by pfraze** [comment]
Related: https://github.com/ssbc/secure-scuttlebutt/pull/135

**Comment by dominictarr** [comment]
it looks like you are running the regexp over the raw data - it would be better to traverse the parsed js object and then test each thing. checkout ssb-msgs#links https://github.com/ssbc/ssb-msgs/blob/master/index.js#L112-L120

**Comment by NHQ** [comment]
does not this module work on strings of text?

**Comment by dominictarr** [comment]
@NHQ hmm I guess the links could be in markdown text... that isn't as bad... what kind of text are the links embedded in?

**Comment by NHQ** [comment]
"data" is text, no matter what kind, markdown or plain.  The new functions
don't validate links, per se, they pluck them from texts.  The intended use
is between the textarea and the sbot post, for parsing plain texts and
getting ssb links.

**Comment by pfraze** [comment]
That seems reasonable. Im going to post some code-review comments

File: index.js [hunk 1]
```diff
@@ -8,6 +8,78 @@ function isString(s) {
...
+  function (data) {
+    return /(%[A-Za-z0-9\/+]{43}=\.sha256)/.exec(data)
+  }
+
+var hasBlobId = exports.hasBlob = exports.hasBlobId =
+  function (data) {
+    return /(&[A-Za-z0-9\/+]{43}=\.sha256)/.exec(data)
+  }
+
+var hasTagId = exports.hasTag = exports.hasTagId =
```
Comment (pfraze) [inline index.js]: The tag functions presuppose we're going to introduce tags. If they stay in, this PR will have to wait for us to decide on tags.

File: package.json [hunk 2]
```diff
@@ -15,8 +15,7 @@
     "tape": "~4.0.1"
   },
   "scripts": {
-    "test": "node test/index.js",
-    "prepublish": "npm ls && npm test"
```
Comment (pfraze) [inline package.json]: Not sure prepublish should be removed, definitely its removal shouldnt be mixed into this issue

File: index.js [hunk 3]
```diff
@@ -8,6 +8,78 @@ function isString(s) {
   return 'string' === typeof s
 }
 
+var hasLink = exports.hasLink =
```
Comment (pfraze) [inline index.js]: I'm ok with these functions returning an exec's match results, but if they are going to do that, they should be named something like `extractLink` instead of `hasLink`. Perhaps the plural version, `extractLinks`, will iterate all matches and put them into an array.

File: index.js [hunk 4]
```diff
@@ -8,6 +8,78 @@ function isString(s) {
...
+
+var hasInvite = exports.hasInvite =
+  function (data) {
+    if(!isString(data)) return false
+    var parts = data.split('~')
+    //console.log(parts, isAddress(parts[0]), /^[A-Za-z0-9\/+]{43}=$/.test(parts[1]))
+    return parts.length == 2 && hasAddress(parts[0]) ?  /([A-Za-z0-9\/+]{43}=)/.exec(parts[1]) : false
+  }
+
+var mentionIt = exports.mentionIt =
```
Comment (pfraze) [inline index.js]: This function supposed to extract all valid refs from an input string, right? I think it'd be faster and clearer to use [exec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec) with regexes using a /g global flag to iterate all the matches. Also the single-character variables are a bit hard to follow.

**Comment by pfraze** [comment]
Also, needs some test cases

**Comment by dominictarr** [comment]
yes, add test cases and we'll merge
